rest/route/host_create.go

```diff
@@ -131,3 +137,113 @@ func (h *hostListHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	return gimlet.NewJSONResponse(results)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/hosts/{container_id}/logs
+
+type containerLogsHandler struct {
+	containerID string
+	startTime   string
+	endTime     string
+	tail        string
+
+	sc data.Connector
+}
+
+func makeContainerLogsRouteManager(sc data.Connector) gimlet.RouteHandler {
+	return &containerLogsHandler{sc: sc}
+}
+
+func (h *containerLogsHandler) Factory() gimlet.RouteHandler { return &containerLogsHandler{sc: h.sc} }
+
+func (h *containerLogsHandler) Parse(ctx context.Context, r *http.Request) error {
+	id := gimlet.GetVars(r)["container_id"]
+	if id == "" {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    "must provide 'container_id'",
+		}
+	}
+	host, err := host.FindOne(host.ById(id))
+	if host == nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("Container %s not found", id),
+		}
+	}
+	if err != nil {
+		return gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    "Error loading host for container " + id,
+		}
+	}
+	h.containerID = id
+
+	if startTime := r.FormValue("start_time"); startTime != "" {
+		if _, err := time.Parse(time.RFC3339, startTime); err != nil {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusBadRequest,
+				Message: fmt.Sprintf("problem parsing start time from '%s' (%s). Must be given in the following format: %s",
+					startTime, err.Error(), time.RFC3339),
+			}
+		}
+		h.startTime = startTime
+	}
+	if endTime := r.FormValue("end_time"); endTime != "" {
+		if _, err := time.Parse(time.RFC3339, endTime); err != nil {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusBadRequest,
+				Message: fmt.Sprintf("problem parsing end time from '%s' (%s). Must be given in the following format: %s",
+					endTime, err.Error(), time.RFC3339),
+			}
+		}
+		h.endTime = endTime
+	}
+	if tailStr := r.FormValue("tail"); tailStr != "" {
+		tail, err := strconv.Atoi(tailStr)
+		if (err == nil && tail >= 0) || (err != nil && tailStr == "all") {
+			h.tail = tailStr
+		} else {
+			return gimlet.ErrorResponse{
+				StatusCode: http.StatusBadRequest,
+				Message: fmt.Sprintf("tail %s invalid, must be non-negative integer or 'all'",
+					tailStr),
+			}
+		}
+	}
+
+	return nil
+}
+
+func (h *containerLogsHandler) Run(ctx context.Context) gimlet.Responder {
+	host, err := host.FindOneId(h.containerID)
```

Comment:
didn't we look up the host by ID in the parse statement? can we just stash it in the handler implementation?

Reply:
Sure can do